pkg/clusteragent/autoscaling/externalmetrics/metrics_retriever.go

```diff
@@ -75,7 +75,8 @@ func (mr *MetricsRetriever) Run(stopCh <-chan struct{}) {
 }
 
 func (mr *MetricsRetriever) retrieveMetricsValues() {
-	if mr.splitBatchBackoffOnErrors {
+	// If any metric encountered a Rate Limit error, do not split up the batch
+	if mr.splitBatchBackoffOnErrors && !mr.store.ContainsErrorType(autoscalers.IsRateLimitError) {
```

Comment:
With `ContainsErrorType` call you're already going through all the store at once, while you could check in the second `GetFiltered` if `datadogMetric.Error` is a Rate Limit error or not